tests/src/codegen/cond_brs.rs

```diff
@@ -0,0 +1,99 @@
+//! Test codegen of conditional branches.
+
+use super::{
+    I16_MAX, I16_MIN, I32_MAX, I32_MIN, I64_MAX, I64_MIN, I8_MAX, I8_MIN, U16_MAX, U16_MIN,
+    U32_MAX, U32_MIN, U64_MAX, U64_MIN, U8_MAX, U8_MIN,
+};
+use paste::paste;
+use ykshim_client::{compile_trace, start_tracing, TracingKind};
+
+/// Generates a test for a binary operation.
+macro_rules! mk_cond_br_tests {
+    ($name: ident, $op: tt, $type: ident, $arg1: expr, $arg2: expr) => {
+        paste! {
+            #[test]
+            fn $name() {
+                #[derive(Eq, PartialEq, Debug)]
+                struct CmpCtx {
+                    arg1: $type,
+                    arg2: $type,
+                    res: u8,
+                }
+
+                impl CmpCtx {
+                    fn new(arg1: $type, arg2: $type, res: u8) -> Self {
+                        Self { arg1, arg2, res }
+                    }
+                }
+
+                #[interp_step]
+                fn interp_step(ctx: &mut CmpCtx) -> bool {
+                    if ctx.arg1 $op ctx.arg2 {
```

Comment:
I considered this, but I didn't want the compiler to remove the conditional branch.

It could change:
```
if ctx.arg1 $op ctx.arg2 {
  ctx.res = 1
} else {
  ctx.res = 0
}
```

into:

```
ctx.res = ctx.arg1 $op ctx.arg2;
```

Reply:
Isn't that the same I suggested above?

Reply:
Yes, but the whole point of these tests are to check that conditional branching is working. So we need to test code that will have a conditional branch :)

Reply:
Got you!